src/passes/sourceUnitPathFixer.ts

```diff
@@ -0,0 +1,31 @@
+import fs from 'fs';
+import path from 'path';
+import { SourceUnit } from 'solc-typed-ast';
+import { AST } from '../ast/ast';
+import { ASTMapper } from '../ast/mapper';
+
+export class SourceUnitPathFixer extends ASTMapper {
+  constructor(private includePaths: string[]) {
+    super();
+  }
+
+  visitSourceUnit(node: SourceUnit, _ast: AST): void {
+    if (fs.existsSync(node.absolutePath)) {
+      for (const prefix of this.includePaths) {
+        const filePath = path.join(prefix, node.absolutePath);
+        if (fs.existsSync(filePath)) {
+          node.absolutePath = filePath;
+          break;
+        }
```

Comment:
Shouldn the most outer `if` condition be negated like:
```typescript
if (!fs.existSync(node.absolutePath)) { ... }
```
Would it be very complex to write a test to check this?

Reply:
I agree we should have tests, I was looking at our cli tests and they are just for input encoding/decoding. We don't tests that invoke from the cli

Reply:
The majority of the work has been done in this PR: https://github.com/NethermindEth/warp/pull/662